src/System.Text.RegularExpressions/src/System/Buffers/SpanHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Buffers
+{
+    internal static class SpanHelper
+    {
+        public static string CopyInput(this ReadOnlySpan<char> input, Span<char> destination, bool targetSpan, out int charsWritten)
+        {
+            if (targetSpan)
+            {
+                if (input.TryCopyTo(destination))
+                    charsWritten = input.Length;
+                else
+                    charsWritten = 0;
+
+                return null;
+            }
+            else
+            {
+                charsWritten = 0;
+                return input.ToString();
+            }
+        }
+
+        public static string CopyOutput(this ValueStringBuilder vsb, Span<char> destination, bool reverse, bool targetSpan, out int charsWritten)
```

Comment:
It seems clumsy to have a method that returns either a string (and doesn't use 2 parameters) or null (and uses those parameters)

Why not have two overloads? One that has the last two parameters and returns void, the other that does not and returns string.

Reply:
Same for CopyInput

Reply:
The follow this pattern in other places too in the BCL, e.g. in BigInteger: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L506 done by @stephentoub. IMO this is the easiest way for code sharing without unnecessary duplicate members for both Span and string.

Reply:
@stephentoub opinion?